src/main/java/bio/terra/workspace/service/datareference/model/BigQueryDatasetReference.java

```diff
@@ -0,0 +1,49 @@
+package bio.terra.workspace.service.datareference.model;
+
+import bio.terra.workspace.generated.model.GoogleBigQueryDatasetUid;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.auto.value.AutoValue;
+
+/**
+ * A representation of a reference to a BigQuery dataset.
+ *
+ * <p>The {@Code JsonTypeName} annotation specifies the class name used for serialization (see the
+ * {@Code JsonSubTypes} annotation in {@Code ReferenceObject} for corresponding deserialization). By
+ * using a constant string instead of the actual class name, changing the name of this class will
+ * not break backwards compatibility with existing serialized objects. This string does not need to
+ * match the class name - it only matches for clarity.
+ */
+@AutoValue
+@JsonTypeName("BigQueryDatasetReference")
+public abstract class BigQueryDatasetReference implements ReferenceObject {
+
+  @JsonCreator
+  public static BigQueryDatasetReference create(
+      @JsonProperty("projectId") String projectId,
+      @JsonProperty("datasetName") String datasetName) {
+    return new AutoValue_BigQueryDatasetReference(projectId, datasetName);
+  }
+
+  @JsonProperty("projectId")
+  public abstract String projectId();
```

Comment:
Some of our getters start with `get` and others don't it seems.

Reply:
I thin all of our autovalue classes don't start with `get`, while generated swagger classes do. I'm not thrilled with it and we can change the autovalue classes if we want for consistency, but that might be more hassle than it's worth.

Reply:
I want to re-open the question of whether we should use AutoValue classes at all.
I was neutral on them at first, but they don't work well with Jackson - which means they don't work well as Stairway parameters. That means that we not only do translation from the API classes to internal ones, but we have to do a second explosion/translation to have them work in Stairway. Instead of being simple, they become littered with Json annotations.

Given that IntelliJ can auto-magically generate and regenerate the internals of a value class, I am not seeing a big win.

Reply:
ack. We use AutoValue throughout WM so I think changing it is out of scope here, but I opened PF-433 to track that effort and can pick that up separately.